# Meta_QC

## Abstract
Our goal is to identify errors in assembly creation based on short reads. Examples of errors are: inversions, chimeras (translocation), indels (<50bp), replacements (large substitutions)

## Workflow
![Flowchart](https://github.com/NCBI-Codeathons/Meta_QC/blob/master/figures/workflow1.jpg)

## Methods
### Datasets
+ Zymobiomics Microbial Community Standards Assemblies (10 genomes; 5 gram positive, 3 gram negative, 2 yeast)
  + Short reads (http://ftp-private.ncbi.nlm.nih.gov/nist-immsa/IMMSA/)
  (https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1299-7#MOESM2)
  + Long reads 
+ Shakya Assembly (MegaHIT and MetaSPAdes)

## Key Results
This will be updated after project completion.

## Discussion 

### Future Directions
This will be updated after project completion.

### Reproduction
#### Docker

## Contributors
+ Todd Treangen
+ Michael Jochum
+ Adam English
+ Shakuntala Mitra
+ Junzhou Wang
+ Yongze Yin
+ Advait Balaji
+ Dreycey
